crates/test-macro/src/lib.rs

```diff
@@ -76,7 +94,70 @@ pub fn wasm_bindgen_test(
     tokens.into_iter().collect::<TokenStream>().into()
 }
 
-fn find_ident(iter: &mut token_stream::IntoIter) -> Option<Ident> {
+fn parse_should_panic(
+    body: &mut std::iter::Peekable<token_stream::IntoIter>,
+    token: &TokenTree,
+) -> Option<Option<Literal>> {
+    // Start by parsing the `#`
+    match token {
+        TokenTree::Punct(op) if op.as_char() == '#' => (),
+        _ => return None,
+    }
+
+    // Parse `[...]`
+    let group = match body.peek()? {
+        TokenTree::Group(group) if group.delimiter() == Delimiter::Bracket => group,
+        _ => return None,
+    };
+
+    let mut stream = group.stream().into_iter();
+
+    // Parse `should_panic`
+    match stream.next()? {
+        TokenTree::Ident(token) if token == "should_panic" => (),
+        _ => return None,
+    }
+
+    // We are interested in the `expected` attribute or string if there is any
+    match stream.next() {
+        // Parse the `(...)` in `#[should_panic(...)]`
+        Some(TokenTree::Group(group)) if group.delimiter() == Delimiter::Parenthesis => {
+            stream = group.stream().into_iter();
+
+            // Parse `expected`
+            match stream.next() {
+                Some(TokenTree::Ident(token)) if token == "expected" => (),
+                _ => panic!("malformed `#[should_panic(...)]` attribute"),
```

Comment:
It'd be better if a [`compile_error!`](https://doc.rust-lang.org/std/macro.compile_error.html) was emitted here instead of panicking, ideally with the span of the `#[should_panic]` attached to it.

Reply:
`compile_error!` doesn't support spans. We aren't using `syn` either. I could write our own `compile_error!` implementation that supports spans if you would like?

I am going to convert the other panics in to `compile_error!`s too.

Reply:
Done.